From fb0049c2509a90b23e11395c1a4d3853f46c8ae7 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:13:08 -0400
Subject: [powerpc] powernv: Include asm/smp.h to fix UP build failure

Message-id: <1410545655-205645-560-git-send-email-dzickus@redhat.com>
Patchwork-id: 94452
O-Subject: [RHEL7 PATCH 559/626] powerpc/powernv: Include asm/smp.h to fix UP build failure
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit b2a80878693256ab97e1e3667e24b3ecdc951de1
Author: Shreyas B. Prabhu <shreyas@linux.vnet.ibm.com>
Date:   Fri Jun 6 15:51:05 2014 +0530

    powerpc/powernv: Include asm/smp.h to fix UP build failure

    Build throws following errors when CONFIG_SMP=n
    arch/powerpc/platforms/powernv/setup.c: In function 'pnv_kexec_wait_secondaries_down':
    arch/powerpc/platforms/powernv/setup.c:179:4: error: implicit declaration of function 'get_hard_smp_processor_id'
        rc = opal_query_cpu_status(get_hard_smp_processor_id(i),

    The usage of get_hard_smp_processor_id() needs the declaration from
    <asm/smp.h>. The file setup.c includes <linux/sched.h>, which in-turn
    includes <linux/smp.h>. However, <linux/smp.h> includes <asm/smp.h>
    only on SMP configs and hence UP builds fail.

    Fix this by directly including <asm/smp.h> in setup.c unconditionally.

    Reported-by: Geert Uytterhoeven <geert@linux-m68k.org>
    Reviewed-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Signed-off-by: Shreyas B. Prabhu <shreyas@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/setup.c b/arch/powerpc/platforms/powernv/setup.c
index 8c16a5f..678573c 100644
--- a/arch/powerpc/platforms/powernv/setup.c
+++ b/arch/powerpc/platforms/powernv/setup.c
@@ -35,6 +35,7 @@
 #include <asm/rtas.h>
 #include <asm/opal.h>
 #include <asm/kexec.h>
+#include <asm/smp.h>
 
 #include "powernv.h"
 
-- 
1.7.1